plugins/tasks/misc/src/main/java/com/walmartlabs/concord/plugins/misc/CollectionsTaskV2.java

```diff
@@ -0,0 +1,81 @@
+package com.walmartlabs.concord.plugins.misc;
+
+import com.google.common.collect.Lists;
+import com.walmartlabs.concord.runtime.v2.sdk.DryRunReady;
+import com.walmartlabs.concord.runtime.v2.sdk.Task;
+
+import javax.inject.Named;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Provides utility methods for working with collections
+ */
+@Named("collections")
+@DryRunReady
+public class CollectionsTaskV2 implements Task {
+
+    /**
+     * Concatenates multiple lists into a single list, preserving the order of elements.
+     * Ignores any null lists.
+     *
+     * @param lists one or more lists to concatenate
+     * @return a new list containing all elements from the input lists
+     */
+    @SafeVarargs
+    public static <T> List<T> concat(List<T> ... lists) {
+        return Arrays.stream(lists)
+                .filter(Objects::nonNull)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Concatenates multiple lists into a single set, eliminating duplicate elements.
+     * Ignores any null lists.
+     *
+     * @param lists one or more lists to concatenate
+     * @return a new set containing all unique elements from the input lists
+     */
+    @SafeVarargs
+    public static <T> Set<T> concatAsSet(List<T> ... lists) {
+        return Arrays.stream(lists)
+                .filter(Objects::nonNull)
+                .flatMap(List::stream)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Reverses the order of elements in the given list.
+     *
+     * @param list the list to reverse
+     * @return a new list with the elements in reverse order
+     */
+    public static <T> List<T> reverse(List<T> list) {
+        return Lists.reverse(list);
+    }
+
+    /**
+     * Creates a new list of integers from 0 to the specified size (exclusive).
+     * Each element in the list corresponds to its index.
+     *
+     * @param size the size of the list to create
+     * @return a list of integers from 0 to (size - 1)
+     */
+    public static List<Integer> newList(int size) {
```

Comment:
Usually a function like this is called `range` :-)

Reply:
> Usually a function like this is called `range` :-)

```suggestion
    public static List<Integer> range(int size) {
```